### DISCUSSION TITLE
Magic number of converted image

### DISCUSSION THREAD

**Discussion by sebastian-james** [ImageMagick/ImageMagick#7413] [Help]
### ImageMagick version

7.0.8-45

### Operating system, version and so on

aws lambda

### Description

I'm doing a conversion between jpg, png, and webp. The conversion works fine. I checked logs, no error while converting. The size changes after conversion. The magic number of the converted image is the same as that of the original image format. Is this expected?

I'm using magic numbers to identify the file type of the image.

**Comment by snibgo** [comment]
> The magic number of the converted image is the same as that of the original image format.

How are you measuring this? The "magic number" is a small number of bytes at the start of a *file* that indicates the encoding method. If you are haven't yet both written to a file and read that file, then the magic number from an in-memory image won't have changed.

**Reply by sebastian-james** [reply]
@snibgo 
Thank you for the response.

The conversion is done in AWS lambda and stored in S3.
```
command = ["/opt/bin/magick", original_image, converted_image]
result = run(command, stdout=PIPE, stderr=PIPE, universal_newlines=True)
```

I determine the magic number of the converted image (fetched from S3) by reading the first few bytes using reactJS, which I take as the original format. I determine the type of the converted image using reactJS file.type of Upload component from antd.

**Comment by snibgo** [comment]
You should check that your build of IM can write webp format.
```
magick -list format
```
Expect a line like this:
```
     WEBP* rw+   WebP Image Format (libwebp 1.3.2 [020F])
```
`rw+` means IM can read and write WEBP, with support for multiple images.

**Reply by sebastian-james** [reply]
Mine gives the following:
```
WEBP* rw-   WebP Image Format (libwebp 0.6.1 [020E])
```

**Comment by snibgo** [comment]
Your libwebp is old. I suggest you upgrade it. Similarly your ImageMagick.

**Reply by sebastian-james** [reply]
Thank you for the input. I'll try that and update
